doc/whats_new.rst

```diff
@@ -64,6 +64,9 @@ Changelog
 Bug
 ~~~
 
+- Fix bug where loading epochs with ``preload=True`` and subsequently using :meth:`mne.Epochs.drop_bad` leads to improper data (and ``epochs.selection``) since v0.16.0 by `Eric Larson`_.
+  If your code uses ``Epochs(..., preload=True).drop_bad(...)``, we recommend regenerating these data.
```

Comment:
problem is limited to the case where reject if specified or changed in the drop_bad right?

Reply:
Yes because `Epochs(..., preload=True).drop_bad()` does nothing